lib/Runtime/Library/BuiltIn/IndexOf.js

```diff
@@ -0,0 +1,48 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+"use strict";
+
+(function indexOf(searchElement, fromIndex) {
+
+    var k;
+
+    if (this == null) {
+        throw new TypeError('"this" is null or not defined');
+    }
+
+    var o = this;
+
+    var len = o.length >>> 0;
```

Comment:
Ah, I see now. The implementation actually defines the behavior of this operator should do that conversion. I was focused on how that's sort of a true thing for asm.js but it shouldn't be observable in the result which would be a JS Number. And since Arrays have a max size of UInt32, that coercion (and not ToInt32) is needed.

Learned something today :P

---
In reply to: [111851902](https://github.com/Microsoft/ChakraCore/pull/2845#discussion_r111851902) [](ancestors = 111851902)